### PR TITLE
B2c search endpoints

### PR DESCRIPTION
**Pull request by pyvovarov-s**
CC-16146, CC-16141 search_endpoints

### CODE REVIEW

File: tests/api/b2c/glue/search_endpoints/catalog_search/negative.robot [hunk 1]
```diff
@@ -0,0 +1,16 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+
+# bug https://spryker.atlassian.net/browse/CC-15983
+Search_without_query_parameter
+    When I send a GET request:    /catalog-search?q=
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/search_endpoints/catalog_search/negative.robot:9]: could you please remove q= from here and from b2b tests

File: tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot [hunk 2]
```diff
@@ -0,0 +1,667 @@
...
+    And Response body parameter should be:    [data][0][attributes][valueFacets][5][activeValue]    None
+    And Response body parameter should be:    [data][0][attributes][valueFacets][5][config][isMultiValued]    False
+    #Filters - weight
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][name]    weight
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][localizedName]    Weight
+    And Response should contain the array of a certain size:    [data][0][attributes][valueFacets][6][values]    ${default_weight_qty}
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][activeValue]    None
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][config][isMultiValued]    True
+    #Filters - rating
+    And Response body parameter should be:    [data][0][attributes][rangeFacets][0][name]    price-DEFAULT-EUR-GROSS_MODE
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot:87]: please add it to a variable

File: tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot [hunk 3]
```diff
@@ -0,0 +1,667 @@
...
+    And Response should contain the array of a certain size:    [data][0][attributes][valueFacets][5][values]    ${default_touchscreen_qty}
+    And Response body parameter should be:    [data][0][attributes][valueFacets][5][activeValue]    None
+    And Response body parameter should be:    [data][0][attributes][valueFacets][5][config][isMultiValued]    False
+    #Filters - weight
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][name]    weight
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][localizedName]    Weight
+    And Response should contain the array of a certain size:    [data][0][attributes][valueFacets][6][values]    ${default_weight_qty}
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][activeValue]    None
+    And Response body parameter should be:    [data][0][attributes][valueFacets][6][config][isMultiValued]    True
+    #Filters - rating
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot:86]: this is not rating - this is price filter

File: tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot [hunk 4]
```diff
@@ -0,0 +1,667 @@
...
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][0][type]    catalog-search
+    And Response body parameter should be:    [data][0][attributes][pagination][numFound]    62
+    And Response body parameter should be:    [data][0][attributes][pagination][currentPage]    1
+    And Response body parameter should be:    [data][0][attributes][pagination][maxPage]    6
+    And Response should contain the array of a certain size:    [data][0][attributes][abstractProducts]    ${default_ipp}
+
+##### FILTERING #####
+
+Filter_by_rating_only_min
```
Comment (OKrivtsova) [inline tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot:261]: b2c has filter by price which is not preset in b2b - we need additional tests for price filter

File: tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot (hunk 2 above)
Comment (pyvovarov-s) [reply tests/api/b2c/glue/search_endpoints/catalog_search/positive.robot:87]: but we don't have it as variable for any filter or b2b version, I would suggest to stick to one version the initial b2b was done in this way plus all another filters in b2c

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**